Enable to access dll because it is being used by another process.

### Description

The MAUI repository on net6 used parallel builds to build faster. When moving to .net7.0 we found a couple of issues and had to disable the build in parallel solution wide while we worked on these problems.

One of the last remaining problems only happens when build MauiRazorClassLibrarySample in Release mode and seems to be only on macOS.  It complains it can't copy the MauiRazorClassLibrarySample.dll to the bin folder.


### Steps to Reproduce

1 - Clone MAUI
2- Checkout the main 
3 - Remove the line on [MauiRazorClassLibrarySample.csproj](https://github.com/dotnet/maui/pull/10420/files#diff-fbb5be75b4074ddf2046b5a4b8901c31c41d9d6d68f92193efc577808bf4bb8f) `<BuildInParallel Condition="'$(Configuration)' == 'Release'">false</BuildInParallel>`
4 - Build on release mode on macOS

### Link to public reproduction project repository

https://github.com/dotnet/maui/

### Version with bug

7.0 (current)

### Last version that worked well

6.0.424

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

*

### Did you find any workaround?

Set BuildInParallel to false on MauiRazorClassLibrarySample

### Relevant log output

```shell
Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 2 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 3 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 4 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 5 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 6 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 7 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 8 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 9 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Beginning retry 10 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4752,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.pdb" to "bin/Release/net7.0/MauiRazorClassLibrarySample.pdb". Beginning retry 1 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.pdb' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4752,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.pdb" to "bin/Release/net7.0/MauiRazorClassLibrarySample.pdb". Beginning retry 2 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.pdb' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4752,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.pdb" to "bin/Release/net7.0/MauiRazorClassLibrarySample.pdb". Beginning retry 3 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.pdb' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4752,5): warning MSB3026: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.pdb" to "bin/Release/net7.0/MauiRazorClassLibrarySample.pdb". Beginning retry 4 in 1000ms. The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.pdb' because it is being used by another process.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): error MSB3027: Could not copy "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". Exceeded retry count of 10. Failed.  [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
/Users/builder/azdo/_work/2/s/bin/dotnet/sdk/7.0.100/Microsoft.Common.CurrentVersion.targets(4685,5): error MSB3021: Unable to copy file "obj/Release/net7.0/MauiRazorClassLibrarySample.dll" to "bin/Release/net7.0/MauiRazorClassLibrarySample.dll". The process cannot access the file '/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/bin/Release/net7.0/MauiRazorClassLibrarySample.dll' because it is being used by another process. [/Users/builder/azdo/_work/2/s/src/BlazorWebView/samples/MauiRazorClassLibrarySample/MauiRazorClassLibrarySample.csproj]
    14 Warning(s)
    2 Error(s)
```

--- a/src/BlazorWebView/src/Maui/build/Microsoft.AspNetCore.Components.WebView.Maui.props
+++ b/src/BlazorWebView/src/Maui/build/Microsoft.AspNetCore.Components.WebView.Maui.props
@@ -1,2 +1,5 @@
 ﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <StaticWebAssetsEnabled Condition="'$(_OuterIntermediateOutputPath)' != ''">false</StaticWebAssetsEnabled>
+  </PropertyGroup>
 </Project>

--- a/src/BlazorWebView/src/Maui/build/Microsoft.AspNetCore.Components.WebView.Maui.targets
+++ b/src/BlazorWebView/src/Maui/build/Microsoft.AspNetCore.Components.WebView.Maui.targets
@@ -121,13 +121,4 @@
     </ItemGroup>
   </Target>
 
-  <!-- Microsoft.Android.Sdk.AssemblyResolution.targets _ResolveAssemblies triggers a nested build to
-       resolve publish files in a Maui specific way and that breaks elements depending on the '$(IntermediateOutputPath)'.
-       This target detects we are running inside one of such builds and adjust the relevant paths. -->
-  <Target Name="_AdjustStaticWebAssetPathBaseWhenBuildingWithRid" BeforeTargets="ResolveStaticWebAssetsConfiguration" Condition="'$(_OuterIntermediateOutputPath)' != ''">
-    <PropertyGroup>
-      <_StaticWebAssetsManifestBase>$(_OuterIntermediateOutputPath)</_StaticWebAssetsManifestBase>
-    </PropertyGroup>
-  </Target>
-
 </Project>

--- a/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj
+++ b/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj
@@ -1,5 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
+<Import Condition=" '$(UseMaui)' != 'true' " Project="..\..\src\Maui\build\Microsoft.AspNetCore.Components.WebView.Maui.props" />
+
   <PropertyGroup>
     <TargetFrameworks>$(MauiDeviceTestsPlatforms)</TargetFrameworks>
     <OutputType>Exe</OutputType>

--- a/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj
+++ b/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
-<Import Condition=" '$(UseMaui)' != 'true' " Project="..\..\src\Maui\build\Microsoft.AspNetCore.Components.WebView.Maui.props" />
+  <Import Condition=" '$(UseMaui)' != 'true' " Project="..\..\src\Maui\build\Microsoft.AspNetCore.Components.WebView.Maui.props" />
 
   <PropertyGroup>
     <TargetFrameworks>$(MauiDeviceTestsPlatforms)</TargetFrameworks>

--- a/src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj
+++ b/src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj
@@ -1,5 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
+  <Import Condition=" '$(UseMaui)' != 'true' " Project="..\..\..\BlazorWebView\src\Maui\build\Microsoft.AspNetCore.Components.WebView.Maui.props" />
+
   <PropertyGroup>
     <TargetFrameworks>$(MauiPlatforms);$(_MauiDotNetTfm)</TargetFrameworks>
     <OutputType Condition="$(TargetFramework.Contains('-'))">Exe</OutputType>

